Source/AwsCommonRuntimeKit/io/HostResolver.swift

```diff
@@ -4,11 +4,27 @@
 import AwsCCommon
 import AwsCIo
 
-public class HostResolver {
+/// All host resolver implementation must conform to this protocol.
+public protocol HostResolverProtocol {
+    /// Resolves the address(es) for HostResolverArguments and returns a list of
+    /// addresses with (most likely) two addresses, one AAAA and one A.
+    func resolveAddress(args: HostResolverArguments) async throws -> [HostAddress]
+    /// Reports a failure on an address so that the background cache can accommodate
+    /// the failure and likely not return the address until it recovers.
+    func reportFailureOnAddress(address: HostAddress) throws
+    /// Empties the cache for an address.
+    func purgeCache(args: HostResolverArguments) async throws
+    /// Empties the cache for all addresses.
+    func purgeCache() async throws
```

Comment:
doesn't seem like these last 3 functions should be throwing.

like, how could the user meaningfully respond to any of these failures

Oh, the underlying C functions can report an error.
Maybe this is a good time to change that

Reply:
Thanks, I have updated the Swift API to ignore the error.